src/main/target/STM32H750/target.h

```diff
@@ -41,33 +76,15 @@
 
 #define SERIAL_PORT_COUNT       (UNIFIED_SERIAL_PORT_COUNT + 8)
 
-#define USE_SPI_DEVICE_1
-#define USE_SPI_DEVICE_2
-#define USE_SPI_DEVICE_3
-#define USE_SPI_DEVICE_4
-#define USE_SPI_DEVICE_5
-#define USE_SPI_DEVICE_6
-
 #define TARGET_IO_PORTA 0xffff
 #define TARGET_IO_PORTB 0xffff
 #define TARGET_IO_PORTC 0xffff
 #define TARGET_IO_PORTD 0xffff
 #define TARGET_IO_PORTE 0xffff
 #define TARGET_IO_PORTF 0xffff
 
-#define USE_I2C
-#define I2C_FULL_RECONFIGURABILITY
-
 #define USE_BEEPER
 
-#ifdef USE_SDCARD
-#define USE_SDCARD_SPI
-#define USE_SDCARD_SDIO
-#endif
-
-#define USE_SPI
```

Comment:
Any reason full reconfigurability cannot be utilised?

If it is due to needing SPI for card storage / flash to boot, we should look into being able to specify pin mappings here, and the remaining pins are reconfigurable - i.e. partial reconfigurability.

Reply:
Noting it is provided for further up the file, if not passed on command line.

Reply:
> Any reason full reconfigurability cannot be utilised?

yes, it's because SPI init is done before config has loaded because the config might be on an external SPI flash or an external QSPI/OCTOSPI flash or on an SD card.  if you don't define the SPI instances in the `.config` file then `spiDevices` array will have each `instance` property set to `NULL` and then all future spi interactions will fail as no instances are defined.

Reply:
In this case you know which one it is for the config... so just define that which is actually required for your target.

Reply:
that's now how the code in `init.c` works, it's not currently possible and is beyond the scope of this PR.